### PR TITLE
travis ci integration 

### PR DESCRIPTION
**Pull request by RiceAndMeet**
- automated build 
- run automated test 
- deploy binary artifacts on tagged commit

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - export PATH="${HOME}/go/bin:$PATH"
+  - sudo add-apt-repository ppa:jonathonf/python-3.6
+  - sudo apt-get update && sudo apt-get install python3.6
+  - pip3 install ipaddr
+  - sudo apt-get install software-properties-common
+  - sudo apt-add-repository --yes --update ppa:ansible/ansible
+  - sudo apt-get install ansible
+install:
+  - cd {TRAVIS_BUILD_DIR}/ansible
+  - ansible-playbook main.yml --extra-vars "ip=35.190.178.102 reset_network=true"
```
Comment (pav-kap) [inline .travis.yml]: Variables like this ip address should be passed via command line or through a separate parameter file

File: ansible/fullnode.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        [Unit]
+        Description=fullnode server
+        After=network.target
+
+        [Service]
+        WorkingDirectory={{GOPATH}}
+        ExecStart=/usr/local/bin/olfullnode node --root {{GOPATH}}\
+          --tendermintRoot {{GOPATH}}/consensus\
+          --seeds 69cbb82a35f621b7a07cfdf9b30b42b943424daa@seed1.chronos.oneledger.network:26615,008a082847ac0d631c174fc23dfd5c2adbe422fa@seed2.chronos.oneledger.network:26615,74fae3565f3d542d0833b7d5cde02d095d6c979f@seed3.chronos.oneledger.network:26615,b36da7eeeff531bb8cac995d8b818b28ea4594e5@seed4.chronos.oneledger.network:26615\
+          --persistent_peers 69cbb82a35f621b7a07cfdf9b30b42b943424daa@seed1.chronos.oneledger.network:26615,008a082847ac0d631c174fc23dfd5c2adbe422fa@seed2.chronos.oneledger.network:26615,74fae3565f3d542d0833b7d5cde02d095d6c979f@seed3.chronos.oneledger.network:26615,b36da7eeeff531bb8cac995d8b818b28ea4594e5@seed4.chronos.oneledger.network:26615 >>\
```
Comment (pav-kap) [inline ansible/fullnode.yml]: Same comment as above for hardcodings here

File: .travis.yml (hunk 1 above)
Comment (RiceAndMeet) [reply .travis.yml]: I don't think you are looking at the latest changes

File: ansible/fullnode.yml [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        ETHAddress = "NONE"
+  - name: create a systemd monitor service file
+    copy:
+      dest: "/etc/systemd/system/olmonitor.service"
+      owner: "root"
+      group: "root"
+      mode: 0644
+      content: |
+        [Unit]
+        Description=montor
```
Comment (reblws) [inline ansible/fullnode.yml]: Mispelled monitor

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+---
+language: go
+
+go:
+  - "1.10.x"
```
Comment (reblws) [inline .travis.yml]: 1.11

File: ansible/fullnode.yml [hunk 5]
```diff
@@ -0,0 +1,121 @@
+- hosts: "{{ip}}"
+  environment:
+    GOPATH: "{{lookup('env','HOME')}}/go"
+  vars:
+    app_user: "{{hostvars[groups['fullnode'][0]]['app_user']}}"
+    GOPATH: "/home/{{app_user}}/go"
+  remote_user: steven
```
Comment (reblws) [inline ansible/fullnode.yml]: ?

File: ansible/fullnode.yml [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+      owner: "root"
+      group: "root"
+      mode: 0644
+      content: |
+        [Unit]
+        Description=fullnode server
+        After=network.target
+
+        [Service]
+        WorkingDirectory={{GOPATH}}
```
Comment (reblws) [inline ansible/fullnode.yml]: change GOPATH to OLDATA

File: ansible/fullnode.yml [hunk 7]
```diff
@@ -0,0 +1,121 @@
+- hosts: "{{ip}}"
+  environment:
+    GOPATH: "{{lookup('env','HOME')}}/go"
+  vars:
+    app_user: "{{hostvars[groups['fullnode'][0]]['app_user']}}"
+    GOPATH: "/home/{{app_user}}/go"
```
Comment (reblws) [inline ansible/fullnode.yml]: Add OLDATA var

File: ansible/fullnode.yml [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      group: "root"
+      mode: 0644
+      content: |
+        [Unit]
+        Description=montor
+        After=network.target
+
+        [Service]
+        WorkingDirectory={{GOPATH}}
+        ExecStart=/usr/local/bin/olmonitor start olvm execute --root {{GOPATH}} >> {{GOPATH}}/olvm.log 2>&1 &
```
Comment (reblws) [inline ansible/fullnode.yml]: Don't redirect output to a log, that should be handled by journald. Remove this text `>> {{GOPATH}}/olvm.log 2>&1 &`

File: ansible/fullnode.yml [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+      owner: "root"
+      group: "root"
+      mode: 0644
+      content: |
+        [Unit]
+        Description=montor
+        After=network.target
+
+        [Service]
+        WorkingDirectory={{GOPATH}}
```
Comment (reblws) [inline ansible/fullnode.yml]: Add a `User=` parameter here so it doesn't run as root

File: genesis.json [hunk 10]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (alexjlan) [inline genesis.json]: We should not put genesis file and config.toml under root folder of the repo, put then under protocol/tests

File: ansible/fullnode.yml [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+      dest: "/etc/systemd/system/olfullnode.service"
+      owner: "root"
+      group: "root"
+      mode: 0644
+      content: |
+        [Unit]
+        Description=fullnode server
+        After=network.target
+
+        [Service]
```
Comment (reblws) [inline ansible/fullnode.yml]: Raise the file limit under here
e.g.
`LimitNOFILE=1000000`

File: ansible/fullnode.yml [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+      group: "{{app_user}}"
+  - name: create config.toml file with content
+    vars:
+      internal_ip: "{{ ansible_all_ipv4_addresses | ipaddr('private') | first }}"
+    copy:
+      dest: "{{GOPATH}}/config.toml"
+      owner: "{{app_user}}"
+      group: "{{app_user}}"
+      content: |
+        Identity = ""
```
Comment (reblws) [inline ansible/fullnode.yml]: Set Identity and NodeName to some value

File: .travis.yml [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+    update: true
+    package:
+    - python
+    - python-pip
+before_install:
+- export OLDATA=$GOPATH/test
+- export OLROOT=$GOPATH/src/github.com/Oneledger
+- export OLSCRIPT=$OLROOT/protocol/node/scripts
+- export OLSETUP=$OLROOT/protocol/node/setup
+- export OLTEST=$OLROOT/protocol/node/scripts
```
Comment (reblws) [inline .travis.yml]: tests

**Pull request closed without merging**